packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart

```diff
@@ -62,11 +62,21 @@ abstract class SharedPreferencesStorePlatform extends PlatformInterface {
   /// * Value type "StringList" must be passed if the value is of type `List<String>`.
   Future<bool> setValue(String valueType, String key, Object value);
 
-  /// Removes all keys and values in the store.
+  /// Removes all keys and values in the store with default prefix.
   Future<bool> clear();
 
+  /// Removes all keys and values in the store with given prefix.
+  Future<bool> clearWithPrefix(String prefix) {
+    throw UnimplementedError('clearWithPrefix is not implemented.');
+  }
+
   /// Returns all key/value pairs persisted in this store.
```

Comment:
This needs a comment like the above as well; it's no longer accurate as written.